cmd/sonobuoy/app/args.go

```diff
@@ -160,6 +166,18 @@ func GetE2EConfig(mode ops.Mode, flags *pflag.FlagSet) (*ops.E2EConfig, error) {
 		}
 		cfg.Parallel = parallel
 	}
+	if flags.Changed(e2eCustomRegistryFlag) {
+		repoFile, err := flags.GetString(e2eCustomRegistryFlag)
+		if err != nil {
+			return nil, errors.Wrap(err, "couldn't retrieve registry list flag")
+		}
+		contents, err := ioutil.ReadFile(repoFile)
+		if err != nil {
+			return nil, errors.Wrap(err, "couldn't read registry list")
+		}
+		cfg.CustomRegistries = string(contents)
```

Comment:
Are all values required or can I just override one of the params and the defaults are used for the others? I wonder if we should add some sort of validation to make sure users don't pass a bad config?

Reply:
I wondered about the validation too but thought that passing the plain file itself just had the benefit of ensuring the implementations were too coupled. We could add a simple yaml unmarshal to ensure its valid yaml at least.

And the code for its consumption is here: https://github.com/kubernetes/kubernetes/blob/master/test/utils/image/manifest.go#L58-L81

It just unmarshals into a map; so if settings aren't provided in the custom file it should be using the defaults. I'll add that to the readme as I'm sure that will be a common question.

Reply:
I think ideally we do the validation upstream, so what you have is good.

Reply:
TL;DR; I'm going to add it :)

Longer version:
eh I'm just split. I just had another issue #615 come up which having a bit of client validation would be nice. Otherwise for users of sonobuoy that means having to 
 - run the tests
 - see the failure
 - pull the logs
 - untar
 - know where to look
 - find the error about parsing
 - loop back to see what was being parsed and hopefully figure out the small repo config was the problem

In both the regexp and yaml cases, we just need to do a yaml/regexp parse and error check. Since we already document they should be yaml/regexp we are only slightly increasing the coupling by doing minimal validation clientside.